src/earn/hooks.ts

```diff
@@ -94,3 +92,68 @@ export function useDepositEntrypointInfo({
   }, [asyncExchanges.result])
   return { hasDepositToken, hasTokensOnSameNetwork, hasTokensOnOtherNetworks, canCashIn, exchanges }
 }
+
+export function usePrepareDepositAndWithdrawTransactions(
+  mode: Extract<EarnActiveMode, 'deposit' | 'withdraw' | 'swap-deposit'>
```

Comment:
When i read these functions, I think to myself
> Why are deposit, withdraw, and swap-deposit in one function and claim-rewards, exit, and withdraw in another function?
> Why are withdraw in both functions and why does one call prepareWithdrawTransactionsWithSwap and the other calls  prepareWithdrawTransactions.

The short answer to both of these, i believe, is because the two functions are used in two different screens and the grouping is purely based on UX choices which is very hard to determine by just looking at these functions. Can they be re-named to be more closely associated with the screens they are used in and maybe comments added to help explain why these two functions exist?

Reply:
@jh2oman That's correct, the grouping is based on which screen they are used on, and withdraw is used on both the `EarnEnterAmount.tsx` & `EarnConfrimationScreen.tsx`. I've updated the names and added a short comment on why these separate hooks are needed in 457a5945c8df94df36661f7b934ab0b2c17a4ce1.